### ISSUE TITLE
Unable to Load, Black Screen...

### ISSUE THREAD

**Issue by JamesSunderland21** [elishacloud/Silent-Hill-2-Enhancements#92]
Dear Team,

SH2 is in my opinion the best game ever so I was very happy to learn about your incredible optimization of the game. I have a physical PC version of Silent Hill 2 Director's Cut (European version). I carefully tried to install the game 6 times, all day long... I have the exact problem than ticket #29. 

I get a blank, black screen with a small text at the top that reads:

"ReShade 3.4.0 345
Visit https://reshade.me for news, updates, shaders and discussion.
Loading (4 effects remaining) … This might take a while. The application could become unresponsive for some time."

It loads forever and I have to reboot my computer each time... So frustrating : (

I tried #29's solution (installation via USB DD) but it didn't work for me. 

I use a Dell XPS computer and Windows 10.

Can you please help me? Thank you.

**Comment by AeroWidescreen** [comment]
If you can't get ReShade to work then you could always remove it / skip that part of the installation. It's a not a crucial thing, unlike the rest of the enhancements.

**Comment by Polymega** [comment]
Hi @JamesSunderland21 ,
Do me a favor and try out this new ReShade package. It's updated to v 4.1.0. Let me know if that resolves the hang-ups for you when you launch. 

1. **Delete** the following files/folders from the game's installation directory, if you haven't already done so: 
   - `\reshade-shaders\`
   - `d3d9.dll`
   - `d3d9.ini`
   - `d3d9.log`
   - `ReShade Filter - SH2 PC PS2.ini`
2. [Download the new ReShade package. ](http://www.enhanced.townofsilenthill.com/SH2/files/SH2PC_ReShade_4.1.0_DirectX9.zip)
   - Place the files in your game's installation directory
3. Ensure `d3d8to9 =` is set to `1` in the `d3d8.ini` file.

**If the game still hangs on launch** after updating ReShade then please attach your `d3d8.log` file here.

The next time we release an update for SH2:EE that newer version of ReShade will also be publicly released with it; so people can download all the new updates in one go.

**Comment by JamesSunderland21** [comment]
Thank you @Polymega and @AeroWidescreen for your help.

@Polymega 
I tried your solution but there is the same issue... 
When I remove all the ReShade files, I have a black screen (only) that remains inactive.

Here is my d3d8.log file.
I'm also attaching my SH2 installation file content.
[d3d8.log](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/2897769/d3d8.log)
![my sh2 file](https://user-images.githubusercontent.com/47930261/53297703-7d13bd80-3822-11e9-9722-cc4a5e5aa212.PNG)

Many thanks.

**Comment by JamesSunderland21** [comment]
When I installed the SH2 Enhancements (Release v1.2.1289.0), I didn't download/touch the "source codes" and only copied & pasted the d3d8.zip content into my folder.

**Comment by JamesSunderland21** [comment]
I tried the English application file and I get this:

**Comment by JamesSunderland21** [comment]
![img_20190224_173459](https://user-images.githubusercontent.com/47930261/53302120-29719600-385b-11e9-8d92-57cfcd80d78d.jpg)
(sorry I previously closed the topic by mistake)

**Comment by JamesSunderland21** [comment]
Here is what I get with the Multilanguage file (SH2PC_Enhanced_EXE_Multilanguage_DC):

**Comment by JamesSunderland21** [comment]
![img_20190224_174503](https://user-images.githubusercontent.com/47930261/53302216-730eb080-385c-11e9-8d7d-b0c6cdbcb67b.jpg)

**Comment by Polymega** [comment]
Try this for me: 

Open the `d3d8.ini` file and set `AutoUpdateModule =` to `0`.

Save changes then try relaunching the game again.

**Comment by JamesSunderland21** [comment]
It doesn't work @Polymega 😔

**Comment by Polymega** [comment]
Oh, did you try removing all the ReShade files and see if that works? Here are the files you will remove: 

* `\reshade-shaders\`
* `d3d9.dll`
* `d3d9.log`
* `ReShade.ini`
* `ReShade Filter - SH2 PC PS2.ini`

And if that doesn't work, open the `d3d8.ini` file and set `d3d8to9 =` to `0`. Save changes and try re-launching the game.

**Comment by JamesSunderland21** [comment]
When I do this, the screen is black and nothing happens. @Polymega

**Comment by JamesSunderland21** [comment]
(thank you for your help)

**Comment by Polymega** [comment]
To confirm: Did you also set `d3d8to9 =` to `0` and try that? 

If so, my last recommendation for now would be to launch the game and let it sit for up to 120 seconds. What we're doing here is trying to see if the game will eventually load if we wait a while.

**Comment by JamesSunderland21** [comment]
Yes, I did (several times).
Nothing occurs after 120 seconds...

**Comment by JamesSunderland21** [comment]
I must have done something wrong during the install... but I've been careful and even watched some videos on YouTube yesterday... I don't understand... This is so frustrating...

**Comment by JamesSunderland21** [comment]
Is there any file missing in my folder?

**Comment by AeroWidescreen** [comment]
So even when you remove ReShade it still doesn't work? Because if so then I would suggest you start over from scratch. See if the game works without any mods.

But wait for what Polymega says first. Maybe he has something else you could try.

**Comment by JamesSunderland21** [comment]
No, still not... Yes, the game works without any mods, I tried yesterday.

**Comment by Polymega** [comment]
Sorry you're having so much trouble. 

So this is going to be time-consuming and not fun, but, I would disable each fix inside the `d3d8.ini` file (by setting them to `0`) **one by one,** launching the game **each time** you disable a fix. 

What you would be doing here is a process of elimination until, hopefully, the game launches for you. If/when the game launches, it will tell you what fix is causing the problem. 

One thing I can tell you is my Surface laptop uses Intel HD graphics (the same as your Dell XPS laptop, I believe?), and my laptop **can't** use Reshade with it.

**Comment by JamesSunderland21** [comment]
OK @Polymega I am going to try that. Btw, my Computer is an XPS 8900 (i7-6700). I also have an XPS 13 Laptop but it doesn't have a CD drive.

**Comment by JamesSunderland21** [comment]
@Polymega I've reinstalled everything and replaced each 1 by a 0 inside the d3d7.ini file... Black frozen screen again... 😔

**Comment by AeroWidescreen** [comment]
But when you delete **d3d8.dll,** the game works? If so, do you have any anti-virus software running? Maybe it's being blocked by that somehow.

**Comment by Polymega** [comment]
Looking over your directory, all your files seem to be in order. 

Try @AeroWidescreen 's suggestion of ensuring your anti-virus/-malware software isn't preventing the game from running. 

Also, @elishacloud , here is their d3d8.log file: 

```
14596 10:31:49.694 Starting Silent Hill 2 Enhancements! v1.2.1298.0
14596 10:31:49.694 Dell Inc. XPS 8900 (Desktop)
14596 10:31:49.694 Dell Inc. 0XJ8C4 A00 (Desktop)
14596 10:31:49.694 "NVIDIA GeForce GTX 745"
14596 10:31:49.694 Windows 10 64-bit (10.0.17763) 
14596 10:31:49.694 Running from: "C:\Users\antoi\Documents\KONAMI\Silent Hill 2\sh2pc.exe"
14596 10:31:49.695 Reading config file: "C:\Users\antoi\Documents\KONAMI\Silent Hill 2\d3d8.ini"
14596 10:31:49.695 Loading as dll: "d3d8.dll"
14596 10:31:49.697 Wrapper created for "d3d8.dll"
14596 10:31:49.999 Hooking the FileSystem APIs...
14596 10:31:50.052 Setting SingleCoreAffinity...
14596 10:31:50.052 Bypassing CD check...
14596 10:31:50.052 Finding SFX file locations...
14596 10:31:50.190 Updating SFX memory address locations...
14596 10:31:50.190 Allocating memory buffer for sddata.bin file...
14596 10:31:50.190 Setting PS2 Style Noise Filter...
14596 10:31:50.190 Increasing the Draw Distance...
14596 10:31:50.191 Setting Cemetery Lighting Fix...
14596 10:31:50.191 Setting Rowboat Animation Fix...
14596 10:31:50.191 Updating the Catacomb Meat Cold Rooms color...
14596 10:31:50.191 Hiding Red Cross health indicator during cutscenes...
14596 10:31:50.191 Loading the "Nemesis2000 Fog Fix" module...
14596 10:31:50.303 Loading the "WidescreenFixesPack and sh2proxy" module...
14596 10:31:50.741 Loading the "modupdater" module...
14596 10:31:50.741 LoadModuleFromResourceToFile Error: "modupd15052 10:31:51.514 Redirecting 'Direct3DCreate8' ---> Passing on to 'Direct3DCreate9'
15052 10:32:08.326 Unhooking library functions
15052 10:32:08.326 Reseting screen resolution
15052 10:32:08.346 Unloading Silent Hill 2 Enhancements!
--> Passi6380 10:32:08.103 Unhooking library functions
6380 10:32:08.104 Reseting screen resolution
6380 10:32:08.125 Unloading Silent Hill 2 Enhancements!
```

I thought the issue may have been related to the modupdater due to the line below, but that's apparently not the case: 

`14596 10:31:50.741 LoadModuleFromResourceToFile Error: "modupd15052 10:31:51.514 Redirecting 'Direct3DCreate8' ---> Passing on to 'Direct3DCreate9'`

**Comment by JamesSunderland21** [comment]
I checked, the problem does not come frome my anti-virus (unfortunately)

**Comment by elishacloud** [comment]
@JamesSunderland21, Can you try the update below and let me know what happens?  Just extract these files to the root of the "Silent Hill 2" folder.  Don't change any settings on them yet.  This should put the module into pass-through mode, with nothing enabled.

[d3d8.zip](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/2898616/d3d8.zip)

**Comment by JamesSunderland21** [comment]
Hi @elishacloud Black screen, nothing happens...
By the way, I re-installed everything in English instead of French, but still the same problem.

**Comment by JamesSunderland21** [comment]
It seems that something is avoiding the game to launch but I don't know what... I've tried to launch the game with CD1 inside the CD driver, then without it. I managed to do the update but there is still the same problem. The SH2 DC physical version I bought a week ago is German, but European versions are supposed to work. Here are my d3d8 files, if needed...
[Silent Hill 2.zip](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/2902161/Silent.Hill.2.zip)

**Comment by JamesSunderland21** [comment]
And my d3d9 files... 
[d3d9.zip](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/2902174/d3d9.zip)

**Comment by Polymega** [comment]
This is a Hail Mary attempt, but try this: 

[First, show extensions for file types so you can know for sure what you're editing/moving/deleting.](https://www.thewindowsclub.com/show-file-extensions-in-windows)

Next, **delete** the following files and folders from your directory, if you haven't already done so: 

- `\reshade-shaders\` (folder)
- `d3d9.ini` (file)

Then, download this ZIP package: [TEST.zip](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/2902227/TEST.zip)

Extract its contents to the game's directory, overwriting any previous files if prompted. 

This contains my copies of the SH2 Enhancements mod, ReShade, and their config (INI) files. I'm hoping maybe there was a typo in one of the config files or the like and this will fix it. I'm doubtful this will fix it, but it's worth a try. Because here is where your d3d8.log file ends: 

`12848 19:41:05.518 Redirecting 'Direct3DCreate8' ---> Passing on to 'Direct3DCreate9'`

I believe that's a d3d8to9 issue.

**Comment by JamesSunderland21** [comment]
Thanks @Polymega but it freezes after showing up the same black screen
![capture](https://user-images.githubusercontent.com/47930261/53365400-70d25200-3941-11e9-867f-5d4fb651c420.PNG)
... This is so depressing...

**Comment by Polymega** [comment]
If you **temporarily remove** `dsound.dll` from the directory does the game launch then?

**Comment by JamesSunderland21** [comment]
Now that is interesting @Polymega It launched! It works in 4:3 format (I don't understand why). It crashed later and showed the following message.
![capture](https://user-images.githubusercontent.com/47930261/53368046-0244c280-3948-11e9-8872-6f24922a68fc.PNG)
The SH2 Menu is in French which is a bit weird because I installed it in English.

**Comment by JamesSunderland21** [comment]
OK it seems to work. Only the Menu is in 4:3 ratio format. The game seems to work in 16:9 nicely. Thank you @Polymega What a huuuuuuuge step!

**Comment by Polymega** [comment]
This is progress! Okay, looking at the [Troubleshooting](http://enhanced.townofsilenthill.com/SH2/#troubleshooting) portion of the installation guide: 

Issue | OS | GPU | Recommendation
-- | -- | -- | --
Using IndirectSound freezes/hangs the game upon launch. | Any | Any | [Install DirectX 9 SDK](https://www.microsoft.com/download/details.aspx?id=35)
. | Any | Any | Do not set the value for `earlyStopFadeOutTime =` too high in the `dsound.ini` file.
. | Any | Any | If using audio output from your GPU (such as Nvidia's "HD Audio" component), disable it and use your motherboard's onboard audio output instead.

So the next thing is to download and install the DirectX 9 SDK (link above) and see if the game will launch then. Keep `dsound.dll` **removed for now** from the game's installation directory. If this fixes your issue, we can then try adding `dsound.dll` back afterwards.

**Comment by Polymega** [comment]
> Only the Menu is in 4:3 ratio format.

[Yes, we hope to make the main menu, save screens, and map screens widescreen in the future.](https://github.com/elishacloud/Silent-Hill-2-Enhancements/issues/48#issuecomment-464410040)

But install the DirectX 9 SDK and let us know if you still receive errors even after game launch that's shown from your image: 

![image](https://user-images.githubusercontent.com/47930261/53368046-0244c280-3948-11e9-8872-6f24922a68fc.PNG)

**Comment by JamesSunderland21** [comment]
OK! I will keep you updated tomorrow. Thank you very much for your help @Polymega and everyone.

**Comment by elishacloud** [comment]
The error may also be coming from the modupdater.

Steps:
1. Install DirectX 9.0c from [here](https://www.microsoft.com/download/details.aspx?id=35).  This is needed for both the IndirectSound and for d3d8to9.
2. If the error still happens then set `AutoUpdateModule = 0`

**Comment by JamesSunderland21** [comment]
@elishacloud @Polymega i installed DirectX 9.0c SDK and put back the dsound.dll back into the SH2 folder... and it works like a charm!!

**Comment by Polymega** [comment]
Great! Sorry you had so much trouble with it. If you want you use a PS4 controller you can follow these steps: https://github.com/elishacloud/Silent-Hill-2-Enhancements/issues/88#issuecomment-456726505

**Comment by JamesSunderland21** [comment]
Thanks a lot @Polymega Yes, I followed Seagull's pieces of advice. I'm having a problem with the buttons compatibility right now... L1 can only serve to open the map... That button is locked, I don't know why... In addition, I can't use the same button twice despite setting the appropriate mod on 1 in the d3d8 file.

**Comment by Polymega** [comment]
Unfortunately I can't assist with this as I don't use a PS4 controller on PC. @seagull may be able to lend advice. Looking back at seagull's original notes, he mentions to clear out all the button mappings within the game's Options menu. Did you do that?

**Comment by JamesSunderland21** [comment]
I understand, no worries. Actually, I don't understand what it means: "clear out all button mappings"

**Comment by seagull** [comment]
In the game itself you can configure button mappings. You've got a grid with actions on the _y_ axis and buttons on the _x_ axis in the order of keyboard/joypad.
For each action, select the joypad mapping, hit ENTER and then ESCAPE. That'll clear out the joypad button mapping. That way the game will only accept input from the "keyboard", which you're emulating with JoyToKey/Xpadder.

**Comment by JamesSunderland21** [comment]
Thank you very much, Seagull. OK I think I understand. In the game's button configuration, everything has three question marks. On the right of that screen, there are button numbers associated with each game action. So I need to link these numbers with the buttons that I chose, with the JoytoKey program right?

**Comment by seagull** [comment]
Nope, not quite.
You want the game to not recognise _any_ joypad input at all and use keyboard input only.
Map the game to use the same keyboard buttons as the ones you're mapping to joypad buttons using Xpadder. Sh2.exe should not respond on its own to any joypad input.

**Comment by JamesSunderland21** [comment]
For example: in the game screen, "Button 1" corresponds to "Action". In JoyToKey, I've set Button 1 with the "X" bottom of my keyboard which is linked to my "Cross" button of my PS4 pad. Correct? Many thanks (and sorry)

**Comment by AeroWidescreen** [comment]
@JamesSunderland21 
This is what he wants you to do. Notice how everything in the Joypad section is empty. You have to clear each one by pressing Enter key and then the Esc key.

![options](https://i.imgur.com/P2A31gq.png)

**Comment by seagull** [comment]
It's fine. We'll sort it.
So we're still not hearing each other. You need to unmap _all joypad inputs_ from the game.
Like this: https://i.imgur.com/Nmpk7uy.png

Use only the keyboard section.
For example, "ACTION" might be mapped to the "E" key and read "-" for joypad.
Then in JoyToKey, map "E" to the cross button on your DS4.

**Comment by JamesSunderland21** [comment]
OK, I see... I don't have access to that area of the screen... That's why I'm locked

**Comment by JamesSunderland21** [comment]
It works perfectly. Thank you very much everyone.

**Comment by JamesSunderland21** [comment]
I have one last question for you Seagull... Everything works fine with my PS4 pad, with the cable. I've been trying to use the PS4 without it, in bluetooth mode. But JoyToKey doesn't recognize the pad anymore, except for the 2 joysticks. Do you know why? Windows 10 and DS4 Windows recognize the pad but not JoyToKey.

**Comment by seagull** [comment]
No idea. I use a Sony USB adaptor specifically made for DS4 controllers, consider investing in one of those.

**Comment by JamesSunderland21** [comment]
Thank you so much to all the team members that helped to optimize this jewel, the best game ever... Several years ago, I had bought the PS3 re-edition of the game which was disastrous (I never touched it). Your work has been honoring the Silent Team's masterpiece. Congratulations! Thank you!
